### PR TITLE
Adds botanical grenade casings.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Adds botanical grenade casings. These babies are also able to accept plants instead of only beakers. They are locked behind the explosive research tech nodes and are somewhat of a weaker replacement to seperated chemicals.

Also changes how grenades display their contents when scanned with science goggles so its less of a sore to the eyes.

EDIT: Omega weed is now blacklisted

## Why It's Good For The Game

Makes it more convinient for botany to make grenades. Due to the late game nature of this tech, I don't think this will make or break hydroponics. The entire thing was a shower thought but the reception on the discord was fairly positive.

## Changelog
:cl:
add: Adds researchable botanical grenade casings. Put your plants into them and blow up your foes!
/:cl:

### CODE REVIEW

**Comment by Barhandar** [comment]
So basically worse regular casings? You can make glass with botany, and hence, beakers. Also I _think_ botany might be able to make bottles too.

**Comment by py01** [comment]
I believe the point is that they can be mass produced much quicker as you don't need to grind plants into beakers

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 1]
```diff
@@ -31,10 +31,14 @@
...
 	if(user.can_see_reagents())
 		if(beakers.len)
-			to_chat(user, "<span class='notice'>You scan the grenade and detect the following reagents:</span>")
-			for(var/obj/item/reagent_containers/glass/G in beakers)
-				for(var/datum/reagent/R in G.reagents.reagent_list)
-					to_chat(user, "<span class='notice'>[R.volume] units of [R.name] in the [G.name].</span>")
+			to_chat(user, "<span class='notice'>You scan the grenade and detect the following objects:</span>")
+			for(var/obj/item/reagent_containers/G in beakers)
+				to_chat(user, "<span class='notice'>A [G.name]. It contains:.</span>")
+				if(G.reagents.reagent_list.len)
```
Comment (vuonojenmustaturska) [inline code/game/objects/items/grenades/chem_grenade.dm]: you don't need this check before a loop

Comment (nemvar) [reply code/game/objects/items/grenades/chem_grenade.dm]: I need this check somewhere so line 41 can happen if false. I don't see why not here.

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 2]
```diff
@@ -318,7 +322,13 @@
 		addtimer(CALLBACK(src, .proc/prime), det_time)
 	log_game("A grenade detonated at [AREACOORD(DT)]")
 
-
+/obj/item/grenade/chem_grenade/botany //Intended for making plants explode. Kinda sorta replaces seperated chemicals.
+	name = "botanical grenade"
+	desc = "A custom made grenade that can also use plants matter due to its integrated grinder."
```
Comment (ATH1909) [inline code/game/objects/items/grenades/chem_grenade.dm]: plant matter, not plants matter

**Comment by ATH1909** [comment]
> So basically worse regular casings? You can make glass with botany, and hence, beakers. Also I _think_ botany might be able to make bottles too.
> 
> Also, omega weed has 420 reagent capacity and all plants have 100. Late tech or not, this is literally free forever better than bluespace beakers.

Do note that these special grenade casings require titanium to make, so it's not exactly "free".

**Comment by nemvar** [comment]
For clarity. Omega weed is now blacklisted

**Comment by chaoko99** [comment]
Maybe a slightly more distinct sprite would be a good idea. You've used a very dark green.

**Comment by subject217** [comment]
this is functionally just a way to bypass beaker caps *just* for botanists. not real content and will not be used for anything remotely more interesting than murderboning.

**Comment by Farquaar** [comment]
> 
> 
> Maybe a slightly more distinct sprite would be a good idea. You've used a very dark green.

Yeah, I second this. The sprite should definitely look more unique.

**Comment by Barhandar** [comment]
@subject217
The only plant that isn't 50 (100 with densified chems) reagent capacity is the blacklisted omega weed. It bypasses getting materials for beakers, not beaker caps.

**Comment by optimumtact** [comment]
Im not ready to accept this kind of thing right after people just called me a bunch of rude words over seperated chems.

I might be less peed off later in the year or something

**Pull request closed without merging**